packages/utils/tests/react.test.tsx

```diff
@@ -0,0 +1,307 @@
+
+      const el = document.createElement("div")
+
+      act(() => {
+        const mergedRef = result.current
+        mergedRef(el)
+      })
+
+      expect(ref1.current).toBe(el)
+      expect(ref2.current).toBe(el)
+    })
+  })
+
+  describe("useCallbackRef", () => {
+    test("should return a stable callback ref that calls the latest callback", () => {
+      let latestValue = 0
+      const { result } = renderHook(() =>
+        useCallbackRef(() => {
+          latestValue += 1
+        }),
+      )
+      act(() => {
+        result.current()
+      })
+      expect(latestValue).toBe(1)
+    })
+  })
+
+  describe("assignRef", () => {
+    test("should assign a value to a ref object", () => {
+      const refObject = { current: null }
+      assignRef(refObject, 123)
+      expect(refObject.current).toBe(123)
+    })
+
+    test("should call a ref callback with a value", () => {
+      const refCallback = vi.fn()
+      assignRef(refCallback, 123)
+      expect(refCallback).toHaveBeenCalledWith(123)
+    })
+  })
+
+  describe("mergeRefs", () => {
+    test("should merge multiple refs", () => {
+      const refObject1 = createRef()
+      const refObject2 = createRef()
+      const mergedRef = mergeRefs(refObject1, refObject2)
+      const el = document.createElement("div")
+      mergedRef(el)
+      expect(refObject1.current).toBe(el)
+      expect(refObject2.current).toBe(el)
+    })
+  })
+
+  describe("useUpdateEffect", () => {
+    test("should not call effect on initial render", () => {
+      const effect = vi.fn()
+      renderHook(() => useUpdateEffect(effect, []))
+      expect(effect).not.toHaveBeenCalled()
+    })
+  })
+
+  describe("useAsync", () => {
+    test("should handle async function execution", async () => {
+      const asyncFunction = () => Promise.resolve("test")
+      const { result } = renderHook(() => useAsync(asyncFunction, []))
+      await waitFor(() => {
+        expect(result.current.value).toBe("test")
+      })
+    })
+  })
+
+  describe("useAsyncFunc", () => {
+    test("should return a function that handles async execution", async () => {
+      const asyncFunction = () => Promise.resolve("test")
+      const { result } = renderHook(() => useAsyncFunc(asyncFunction, []))
+
+      let asyncResult: Promise<string>
+
+      await act(async () => {
+        asyncResult = result.current[1]()
+      })
+
+      await waitFor(async () => {
+        await expect(asyncResult).resolves.toBe("test")
+      })
+    })
+  })
+
+  describe("useAsyncRetry", () => {
+    test("should retry async function on retry call", async () => {
+      let attempt = 0
+      const asyncFunction = () => {
+        attempt += 1
+        return attempt === 2
+          ? Promise.resolve("success")
+          : Promise.reject("fail")
+      }
+      const { result } = renderHook(() => useAsyncRetry(asyncFunction, []))
+      expect(result.current.loading).toBeTruthy()
+      await waitFor(() => {})
```

Comment:
Is this intended?

Reply:
Yes, it is. Without this, it will not work.